Angular/Angular guidelines and best practices/Testing.md

```diff
@@ -531,6 +531,96 @@ beforeEach(async(() => {
+
+An alternative approach is to wrap the component you want to test into another component that is declared only in the TestBed and is used only for interacting with the component you want to test via inputs and outputs.
+
+Fixture is created for the host component and `fixture.componentInstance` will not be an instance of the component you want to test, it will be an instance of the host component. If you need the instance of the component you want to test, you can get it after the fixture is initialized, via `debugElement.query` or by adding a `ViewChild` in the host component.
+
+Host component pattern starts to shine when you need to change some input value and trigger CD - you can simply change the public property value on the host component and call `fixture.detectChanges()`. That will trigger `ngOnChanges` and re-render the child component, even if it is using OnPush CD. If you didn't use host component, you would have to manually assign property values and call `ngOnChanges` with the correct `changes` object. That can be tedious to write, or you could even forget to call `ngOnChanges` and then wonder why the tests are not working as expected.
+
+The downside of using a host component is that it is a bit boilerplate-y - you will have to bind potentially many inputs and outputs between the host and child components.
+
+We recommend creating a host component, especially for cases where you need to test input/output interaction.
+
+```typescript
+@Component({
+  selector: 'counter'
+  template: `
+  {{ value }}
+  <button class="counter-button" (click)="onCounterButtonClick()">Increase</button>
+  `
+})
+export class CounterComponent {
+  @Input() public value: number = 0;
+  @Output() public valueChange = new EventEmitter<number>();
+
+  public onCounterButtonClick(): void {
+    this.value++;
+    this.valueChange.emit(this.value);
+  }
+}
+```
+
+```typescript
+@Component({
+  selector: 'counter-host'
+  template: `
+  <counter
+    [value]="value"
+    (valueChange)="onValueChange()
+  >
+  </counter>
+  `
+})
+class CounterHostComponent {
+  public value: number = 0; // Notice that this doesn't have to be an @Input
+
+  public onValueChange(newValue: number) { }
+}
+
+describe('CounterComponent', () => {
+  let fixture: ComponentFixture<CounterHostComponent>;
+  let hostComponent: CounterHostComponent;
+  let component: CounterComponent;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [CounterComponent, CounterHostComponent],
+      imports: [...],
+      providers: [...],
+    }).compileComponents();
+  })
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CounterHostComponent);
+    fixture.detectChanges();
+
+    hostComponent = fixture.componentInstance;
+    component = fixture.debugElement.query(By.css('counter')).componentInstance;
```

Comment:
I would instead use a ViewChild in the CounterHostComponent to create the component instance variable and then use it here.

Reply:
I will change it to ViewChild, but either solution would be fine with me